File: 백준/Gold/5582. 공통 부분 문자열/공통 부분 문자열.py
s1 = '=' + input()
s2 = '-' + input()

len_s1 = len(s1)
len_s2 = len(s2)

arr = [[0 for _ in range(len_s2)] for _ in range(len_s1)]

for i in range(1, len_s1):
    for j in range(1, len_s2):
        if s1[i] == s2[j]:
            arr[i][j] = arr[i-1][j-1] + 1
ans = 0

for line in arr:
    if ans < max(line):
        ans = max(line)

print(ans)

File: 백준/Gold/5582. 공통 부분 문자열/공통 부분 문자열.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main{
	public static void main (String [] args) throws IOException {
	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	String s1 = new String(br.readLine());
	String s2 = new String(br.readLine());
	int len_s1 = s1.length();
	int len_s2 = s2.length();

	int [][] arr = new int [len_s1][len_s2];
	int ans = 0;
	
	for (int i = 0; i < len_s1; i++) {
		for (int j = 0; j < len_s2; j++) {
			if (i==0 || j == 0) {
				if (s1.charAt(i) == s2.charAt(j)) {
					arr[i][j] = 1;
				}
			}
			else {
				if (s1.charAt(i) == s2.charAt(j)) {
					arr[i][j] = arr[i-1][j-1] + 1;
				}
			}
			if (ans < arr[i][j]) {
				ans = arr[i][j];
			}
		}
	}
	System.out.println(ans);
	}
}